pkg/client/unversioned/remotecommand/remotecommand_test.go

```diff
@@ -92,15 +92,21 @@ func fakeExecServer(t *testing.T, i int, stdinData, stdoutData, stderrData, erro
 
 		if len(errorData) > 0 {
 			fmt.Fprint(errorStream, errorData)
+		}
+		if errorStream != nil {
 			errorStream.Close()
 		}
 
 		if len(stdoutData) > 0 {
 			fmt.Fprint(stdoutStream, stdoutData)
+		}
+		if stdoutStream != nil {
 			stdoutStream.Close()
 		}
 		if len(stderrData) > 0 {
 			fmt.Fprint(stderrStream, stderrData)
+		}
+		if stderrStream != nil {
```

Comment:
I touched this in https://github.com/kubernetes/kubernetes/pull/13144

It's actually better to avoid calling Close() on stdoutStream/stderrStream and let the deferred Reset() calls close the streams. Calling Close() explicitly actually introduces a race with the stdinStream read below